### PR TITLE
DM-42528: Fix reporting request status

### PR DESCRIPTION
**Pull request by wguanicedew**
## Checklist

- [ ] ran Jenkins
- [ ] added a release note for user-visible changes to `doc/changes`

### CODE REVIEW

File: python/lsst/ctrl/bps/panda/panda_service.py [hunk 1]
```diff
@@ -179,12 +179,18 @@ def report(
                 WmsStates.PRUNED: "output_missing_files",
             }
 
-            # workflow status to report as SUCCEEDED
-            wf_status = ["Finished", "SubFinished", "Transforming"]
-
-            wf_succeed = False
+            workflow_status = head["status"]["attributes"]["_name_"]
+            if workflow_status in ["Finished", "SubFinished"]:
```
Comment (MichelleGower) [inline python/lsst/ctrl/bps/panda/panda_service.py:183]: What does subfinished mean in particular to understand why it means succeeded?

File: python/lsst/ctrl/bps/panda/panda_service.py [hunk 2]
```diff
@@ -179,12 +179,18 @@ def report(
...
+                wms_report.state = WmsStates.SUCCEEDED
+            elif workflow_status in ["Failed", "Cancelled", "Suspended", "Expired"]:
```
Comment (MichelleGower) [inline python/lsst/ctrl/bps/panda/panda_service.py:185]: WmsStates also has a deleted state (shouldn't that match Cancelled?) and a held state (for Suspended?)

File: python/lsst/ctrl/bps/panda/panda_service.py (hunk 1 above)
Comment (MichelleGower) [reply python/lsst/ctrl/bps/panda/panda_service.py:183]: Does this give the correct status for when the final job hasn't shown up yet in PanDA tasks?

**Review by MichelleGower — APPROVED** [review]
Some questions.  Can be merged after following up on the questions.

File: python/lsst/ctrl/bps/panda/panda_service.py (hunk 1 above)
Comment (wguanicedew) [reply python/lsst/ctrl/bps/panda/panda_service.py:183]: yes. It's the workflow status in iDDS. In the iDDS result, there are two parts about the status: (1) 'status': the workflow/request status, it includes the last task even it's not created. (2) 'transform_status': It's the task status, per task.

In WmsStates, there is no item to distinguish 'Subfinished'. That's why I mapped it to Succeeded.

File: python/lsst/ctrl/bps/panda/panda_service.py (hunk 2 above)
Comment (wguanicedew) [reply python/lsst/ctrl/bps/panda/panda_service.py]: I updated the map.

**Comment by timj** [comment]
The doc build failure is caused by the documenteer version. Please change the github action to pin the documenteer version to 0.8.2.

**Comment by wguanicedew** [comment]
> The doc build failure is caused by the documenteer version. Please change the github action to pin the documenteer version to 0.8.2.

After updating documenteer version to 0.8.2, the build doc still failed. Do you have any idea how to fix it?

**Comment by timj** [comment]
You need to rebase don't you? It's fixed on main via #66 but not on this branch which is still installing a different documenteer version.

**Comment by wguanicedew** [comment]
The build docs work ok. The local test also works ok. I will merge it.

**Pull request merged**